modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -3151,4 +3156,38 @@ else if (plan.hasRows()) {
     public LongRunningQueryManager longRunningQueries() {
         return longRunningQryMgr;
     }
+
+    /**
+     * Register predefined custom aggregate functions.
+     *
+     * @throws IgniteCheckedException If failed.
+     */
+    private void registerAggregateFunctions() throws IgniteCheckedException {
+        registerAggregateFunction(GridFirstValueFunction.NAME, GridFirstValueFunction.class);
+        registerAggregateFunction(GridLastValueFunction.NAME, GridLastValueFunction.class);
+    }
+
+    /**
+     * Register custom aggregate function.
+     *
+     * @param fnName SQL function name.
+     * @param cls Function implementation class.
+     * @throws IgniteCheckedException If failed.
+     */
+    public void registerAggregateFunction(String fnName,
+        Class<? extends AggregateFunction> cls) throws IgniteCheckedException {
+        if (fnName == null)
+            throw new NullPointerException();
```

Comment:
IMO, logging ERROR and silently ignore register request instead of NPE is way better.

Btw, if you so care about argument checking, what if `fnName` would be `''` or `'%'` or reserved SQL word ? or already existing function name?